6609 Erstatt nav frontend skjema

fredrik-oh opened the pull request:
https://jira.adeo.no/browse/MELOSYS-6609

Commit:
Erstatter Checkbox fra nav-frontend-skjema med Aksel Checkbox

Commit:
Erstatter Checkbox fra nav-frontend-skjema med Aksel Checkbox (glemte renaming i index.js)

Commit:
fikser tsc build error

Commit:
Erstatter Input fra nav-frontend-skjema med Aksel TextField

Commit:
TextField. Fikset opp i bruk av !important i .less fil ved å i stedet sende inn en default className med styling i navFrontend/skjema/textfield

janitamarielle commented on src/felleskomponenter/menypanel/menypunkter/virksomhetenINorge/redigerbarSamletVirksomhetINorge.tsx:
```diff
@@ -9,29 +9,25 @@ const RedigerbarSamletVirksomhetINorge = () => {
       <Skjema.Input
         label="Antall ansatte"
         feltNavn="juridiskArbeidsgiverNorge.antallAnsatte"
-        bredde="XS"
```
Ikke sette size small?

fredrik-oh replied:
Defaulter til small

janitamarielle commented on src/felleskomponenter/skjema/listevelger/listevelger.less:
```diff
@@ -30,6 +30,7 @@
 
   .listevelger__linje__input {
     margin-bottom: 0;
+    gap: unset;
```
Aldri sett denne før. Kult.

janitamarielle commented on src/felleskomponenter/ui/checkbox/checkbox.tsx:
```diff
@@ -13,42 +13,24 @@ interface onCheckProperties {
+  };
+
+  return (
+    <Nav.Checkbox onChange={changeHandler} {...rest} id={rest.id ?? Utils._uuid()}>
```
keyPress works out of the box?

fredrik-oh replied:
Jepp

Commit:
Trenger tydeligvis important i textfield.less

Commit:
Oppdaterer snapshots etter TextField endringer

Commit:
Erstatter RadioPanelGruppe med aksel RadioGroup

Commit:
Skriver om Skjema.Radio til å benytte aksel radio + radiogroup

Commit:
Merge branch 'refs/heads/master' into MELOSYS-6609_Erstatt_nav-frontend_skjema

# Conflicts:
#	src/sider/journalforing/komponenter/journalforingform/informasjon.jsx

Commit:
Oppdaterer snapshots og fikser lint- og testfeil

Commit:
Erstattet forms/radio med forms/radiogroup som benytter aksel RadioGroup

Commit:
Erstattet alle forekomster av Nav.Radio med aksel RadioGroup + Radio

janitamarielle commented on src/felleskomponenter/forms/radiogroup.tsx:
```diff
@@ -0,0 +1,52 @@
+          <Nav.RadioGroup
+            {...rest}
+            legend={legend}
+            defaultValue={field.value || defaultValue}
```
?? her også?

janitamarielle commented on src/felleskomponenter/menypanel/menypunkter/arbeidssteder/land/redigererPreElementer.tsx:
```diff
@@ -41,8 +43,10 @@ const RedigererPreElementer = ({ className, redigerbart }: EnRedigeringsknappLis
           <Nav.Column xs="4" className="col">
-            <Skjema.Radio disabled={!redigerbart} label="Ja" feltNavn="arbeidPaaLand.erHjemmekontor" value />
-            <Skjema.Radio disabled={!redigerbart} label="Nei" feltNavn="arbeidPaaLand.erHjemmekontor" value={false} />
+            <Nav.RadioGroup legend="" hideLegend>
```
Trengs legend hvis man uansett skal hideLegend?

janitamarielle commented on src/felleskomponenter/sideDialog/sendBrev/brevMottaker/brevMottaker.tsx:
```diff
@@ -188,22 +188,22 @@ const BrevMottaker = ({
+                    <Skjema.Radio
+                      feltNavn="arbeidsgiver"
+                      label={`${virksomhet.mottakerNavn} (org.nr. ${virksomhet.orgnr})`}
+                      id={`arbeidsgiver.${virksomhet.orgnr}`}
```
key var ikke nødvendig?

fredrik-oh replied:
key skal settes på toppnivå under en mapping av jsx.
https://react.dev/learn/rendering-lists#why-does-react-need-keys

janitamarielle commented on src/felleskomponenter/sideDialog/sendBrev/valgAlternativer.tsx:
```diff
@@ -1,34 +1,32 @@
+            feltNavn={`felt.${feltKode}.valg`}
+            label={alternativ.beskrivelse}
+            id={`${feltKode}.${alternativ.kode}`}
+            key={`${feltKode}.${alternativ.kode}`}
```
Her er key, mens ikke over. Why?

fredrik-oh replied:
se svar

janitamarielle commented on src/felleskomponenter/skjema/customRadioPanelGruppe/customRadioPanelGruppe.jsx:
```diff
@@ -102,21 +95,26 @@ const CustomRadioPanelGruppe = (props) => {
             value={radio.value}
             checked={currentCheckedValue === radio.value}
             {...radio}
             notify={notify}
```
Skal du passe notify inn selv om du dro den ut i onChangeAndNotify?

janitamarielle commented on src/felleskomponenter/skjema/customRadioPanelGruppe/customRadioPanelGruppe.jsx:
```diff
@@ -34,27 +34,22 @@ class CustomRadioPanel extends Component {
       <Fragment>
         <label className={cls} htmlFor={`${feltNavn}-${value}`}>
-          <Nav.Radio
+          <Nav.AkselRadio
```
AkselRadio og ikke bare Radio?

fredrik-oh replied:
Skal rename :)

janitamarielle commented on src/felleskomponenter/skjema/customRadioPanelGruppe/customRadioPanelGruppe.jsx:
```diff
@@ -34,27 +34,22 @@ class CustomRadioPanel extends Component {
             value={value}
             onFocus={() => this.toggleOutline()}
             onBlur={() => this.toggleOutline()}
-            onChange={onChangeAndNotify}
```
Må du fortsat ha onChange={onChange} kanskje?

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingArtikkel11_4/vurderingArtikkel11_4.tsx:
```diff
@@ -199,41 +197,28 @@ const VurderingArtikkel11_4 = ({
-              konvensjonStorbritanniaToggleEnabled
+                : "11.4 ii - arbeidsgiver i bostedslandet"}
+            </Nav.AkselRadio>
+            <Nav.AkselRadio name="artikkel11" value={ArtikkelValg.ART11_4_1_TIL_VURDERING_12_1}>
```
Name som henger igjen?

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingArtikkel16MottaSvar/vurderingArtikkel16MottaSvar.jsx:
```diff
@@ -92,7 +92,12 @@ const FormKomponent = ({
+            disabled={!redigerbart}
+          >
             <Skjema.Radio
               name="svarFraMyndighetene"
```
name skal ikke opp i RadioGroup dersom det er skjema, eller?

fredrik-oh replied:
Skal ikke være nødvendig

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingBostedsland.jsx:
```diff
@@ -100,23 +100,16 @@ const VurderingBostedsland = (props) => {
+              name="bostedsland"
+              disabled={!redigerbart}
+            >
+              <Nav.AkselRadio value={true} onChange={radioEndringHandler}>
```
onChange?

fredrik-oh replied:
Fjerner

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingBostedsland.jsx:
```diff
@@ -100,23 +100,16 @@ const VurderingBostedsland = (props) => {
-                checked={erNorgeValgt === false}
-                disabled={!redigerbart}
-              />
+            <Nav.RadioGroup
```
defaultValue={erNorgeValgt}?

fredrik-oh replied:
Legger til

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingBostedsland.jsx:
```diff
@@ -131,7 +124,7 @@ const VurderingBostedsland = (props) => {
                 </Nav.Row>
               )}
-            </Nav.Fieldset>
+            </Nav.RadioGroup>
```
Ser ikke ut som om det konsekvent wrappes RadioGroup over alt som tidligere hadde Fieldset, feks her der radio-valgene trigger visning. Hva ønsker vi å fortsette med her?

fredrik-oh replied:
aksel RadioGroup lager sitt eget Fieldset så burde fjernes dersom fieldset i vår kode kun er rundt radioknappene

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingForretningssted/vurderingForretningssted.jsx:
```diff
@@ -181,38 +181,36 @@ const VurderingForretningssted = (props) => {
+      <Nav.Select
+        name="artikkel3_1_vurdering"
+        id="vurdering13_1"
+        label="Velg artikkel"
```
Har denne både Velg artikkel her og Velg artikkel nedendfor? 

Edit: Aah, det er første option. Kanskje bytte til Velg... som er standarden?

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingMedfolgendeBarn.jsx:
```diff
@@ -55,24 +55,20 @@ const MedfolgendeBarn = ({
+          name={radioName}
+          size="small"
+        >
+          <Nav.AkselRadio name={radioName} value={BOOLSK_STRING.SANN}>
```
Fjern name.

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingMedfolgendeBarn.jsx:
```diff
@@ -55,24 +55,20 @@ const MedfolgendeBarn = ({
+          hideLegend
+          disabled={!redigerbart}
+          onChange={onCheck}
+          defaultValue={omfattet}
```
Omfattet er true eller false, mens verdiene nedenfor er TRUE eller FALSE. Må kanskje gjøre true/false om til BOOLSK_STRING, tror det er en funksjon for det i utils.

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingUtpekt/vurderingUtpekt.jsx:
```diff
@@ -205,7 +205,12 @@ export const VurderingUtpekt = ({
+          >
             <Skjema.Radio
               label="Godkjenn"
               forhandsvalgt={!erBucAapen}
```
Skulle forhandsvalgt-logikken heller flyttet seg til defaultValue?

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingUtpekt/vurderingUtpekt.jsx:
```diff
@@ -219,7 +224,7 @@ export const VurderingUtpekt = ({
               name="godkjenn"
```
name her og ikke i RadioGroup?

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingYrkesaktivitet.tsx:
```diff
@@ -73,44 +73,37 @@ const VurderingYrkesaktivitet = (props: VurderingYrkesaktivitetProps) => {
-      <Nav.Fieldset legend="">
-        <Nav.Radio
+      <Nav.RadioGroup
+        legend=""
```
hideLabel?

janitamarielle commented on src/sider/eu_eøs/stegKomponenter/vurderingYrkesaktivitet.tsx:
```diff
@@ -73,44 +73,37 @@ const VurderingYrkesaktivitet = (props: VurderingYrkesaktivitetProps) => {
+        name="yrkesaktivitet"
+        disabled={!redigerbart}
+      >
+        <Nav.AkselRadio name="yrkesaktivitet" value={KV.Koder.VurderingYrkesaktivitetTyper.ORDINAER_ARBEIDSTAKER}>
```
name?

Edit: Dette gjelder mesteparten av radioene nedenfor også.

janitamarielle commented on src/sider/ftrl/saksbehandling/stegKomponenter/vurderingTrygdeavgift/komponenter/skatteforholdsperioder.tsx:
```diff
@@ -57,23 +57,17 @@ export const Skatteforholdsperioder = ({
-                  />
-                </div>
+                <Forms.RadioGroup
+                  legend={index === 0 ? <Nav.Typo.Element>Er bruker skattepliktig?</Nav.Typo.Element> : null}
```
Her bruker du null som legend-verdi, men ellers har du brukt ""

janitamarielle commented on src/sider/journalforing/komponenter/avsender/avsenderVelgerForBruker.tsx:
```diff
@@ -88,18 +89,12 @@ const AvsenderVelgerForBruker = ({
-          value={MKV.Koder.avsendertyper.PERSON}
-          className="avsendervelger__radio"
-        />
+      <Nav.RadioGroup legend="Hvem er avsender?">
```
Ikke noe defaultValue?

janitamarielle commented on src/sider/journalforing/komponenter/avsender/avsenderVelgerForVirksomhet.tsx:
```diff
@@ -25,22 +26,10 @@ export const AvsenderVelgerForVirksomhet = ({
-        <Skjema.Input label="" feltNavn="avsenderNavn" bredde="fullbredde" />
-      )}
-    </Skjema.RadioGruppe>
+    <Nav.RadioGroup legend="Hvem er avsender?">
```
Ingen defaultValue trengs?

janitamarielle commented on src/sider/journalforing/komponenter/fagsakVelger.jsx:
```diff
@@ -70,24 +69,10 @@ const FagsakVelger = (props) => {
-            value={OPPRETT}
-          />
-        </div>
+        <Nav.RadioGroup defaultValue={EKSISTERENDE} onChange={setValgtVisning} legend="" hideLegend>
```
ikke defaultValg=valgtVisning?

janitamarielle commented on src/sider/journalforing/komponenter/fagsakVelger.jsx:
```diff
@@ -70,24 +69,10 @@ const FagsakVelger = (props) => {
+        <Nav.RadioGroup defaultValue={EKSISTERENDE} onChange={setValgtVisning} legend="" hideLegend>
+          <Nav.AkselRadio value={EKSISTERENDE}>{EKSISTERENDE}</Nav.AkselRadio>
+          <Nav.AkselRadio value={OPPRETT}>{OPPRETT}</Nav.AkselRadio>
+        </Nav.RadioGroup>
```
Burde ikke RadioGroup ta over name? Er vel fare for at journalføring har flere radio-knapper?

janitamarielle commented on src/sider/journalforing/komponenter/journalforingGjelder.tsx:
```diff
@@ -67,17 +66,14 @@ const JournalforingGjelder = ({
-          onChange={handleClick}
-          className="journalforingGjelder"
-        />
+        <Nav.RadioGroup onChange={handleClick} defaultValue={journalforingGjelder} legend="" hideLegend>
```
Byebye name? Samme kommentar som over. Journalføring har flere radio-grupper, burde kanskje ha unike names?

janitamarielle commented on src/sider/journalforing/komponenter/knyttTilSak.jsx:
```diff
@@ -156,20 +156,17 @@ export const KnyttTilSak = (props) => {
-          </Skjema.RadioGruppe>
+          <div className="panelElement">
+            <Nav.Typo.Undertittel className="overskrift">Velg hva du vil gjøre med dokumentet</Nav.Typo.Undertittel>
+            <Nav.RadioGroup legend="" hideLegend size="small">
```
defaultValue? Trengs det kanskje ikke for Skjema-radioer?

janitamarielle commented on src/sider/opprettnysak/komponenter/oppgaveVelger.tsx:
```diff
@@ -73,29 +71,22 @@ export const OppgaveVelger = ({
-          value
-        />
+      <Nav.RadioGroup legend="" hideLegend>
+        <Skjema.Radio feltNavn="oppretterOppgave" label="Opprett ny oppgave" name="velgVisningOppgave" value />
```
Name i skjema-radio istedenfor radiogroup

janitamarielle commented on src/sider/opprettnysak/opprettnysak.tsx:
```diff
@@ -299,17 +299,17 @@ const OpprettNySak = ({
-              />
+              <div className="innrykk">
+                <Nav.RadioGroup
+                  defaultValue={BRUKER}
```
name

janitamarielle commented on src/sider/trygdeavtale/stegKomponenter/vurderingAvklarVirksomhet/vurderingAvklarVirksomhet.tsx:
```diff
@@ -78,17 +78,11 @@ const VurderingAvklarVirksomhet = ({
 
       {virksomheterListe?.length !== 0 ? (
-        <Skjema.RadioGruppe feltNavn="virksomhet" label="">
+        <Nav.RadioGroup legend="" hideLegend disabled={!redigerbart} defaultValue={resultat?.virksomhet}>
```
Her har skjema defaultValue, så kan hende det da mangles på endel av komponentene, evt at det ikke trengs her.

janitamarielle commented on src/sider/trygdeavtale/stegKomponenter/vurderingBestemmelse/vurderingBestemmelse.tsx:
```diff
@@ -116,7 +116,7 @@ const VurderingBestemmelse = ({
       <Nav.Typo.Innholdstittel className="stegvelgertittel">Bestemmelse og vurdering</Nav.Typo.Innholdstittel>
 
-      <Nav.Fieldset legend="Hva er din vurdering av søknaden?">
+      <Nav.RadioGroup legend="Hva er din vurdering av søknaden?" defaultValue={formValues?.vedtak}>
```
Har jeg skjønt det rett i at onChangen nedenfor kunne blitt dratt opp hit? Samt det som er felles av disabled-feltet?

janitamarielle commented on src/sider/unntaksregistrering/vurderingUnntakMedlemskap/vurderingUnntakMedlemskap.tsx:
```diff
@@ -179,31 +179,19 @@ const VurderingUnntakMedlemskap = ({ oppdaterStatus, tilbake, aktivtSteg }: Vurd
         <Nav.Column xs="8">
-          <Nav.Typo.Normaltekst className="formLabel">Vurder unntaksperiode</Nav.Typo.Normaltekst>
-          <Forms.Radio
+          <Forms.RadioGroup
```
Når bruker man Forms.RadioGroup vs Nav.RadioGroup + Forms.Radio?

fredrik-oh replied:
Har slettet Forms.Radio

janitamarielle reviewed (commented):
160 filer 🙃

Commit:
Fikser lint og testfeil. Justerte en rekke true/false radiogrupper

Commit:
Justeringer etter QA. Har besluttet å erstatte Skjema.Radio med Skjema.RadioGroup på samme måte som ble gjort for Forms så det kommer en ny runde med endringer.

Commit:
Tok med et par gjenglemte justeringer + oppdaterte snapshot

Commit:
Erstattet Skjema.Radio med ny komponent Skjema.RadioGroup for å samsvare med hvordan omskrivningen ble håndtert for nyere (Forms.RadioGroup) radioknapper.

Slettet Skjema.Radiogruppe

Noe vilkårlig refaktorering av unødvendig rotete løsninger, spesielt i felleskomponenter/menypanel...

janitamarielle commented on src/felleskomponenter/menypanel/menypunkter/ovrigOmArbeidstaker/redigerer/labelOgEditerbartSvar.tsx:
```diff
@@ -1,38 +1,19 @@
-        {svar}
-      </Nav.Column>
-    </fieldset>
+    <Skjema.RadioGroup legend={label} hideLegend name={feltNavn}>
```
legend={label} sammen med hideLegend?

fredrik-oh replied:
Fikser

janitamarielle replied:
Det var flere steder også, så ta gjerne en gjennomgang

Commit:
Fjernet unødvendige hideLegend props fra konsumenter av RadioGroup
